### PR TITLE
VDR refactor part trois: implement SubjectManager interface for nuts and web

### PR DESCRIPTION
**Pull request by woutslakhorst**
part of #3208

This PR adds the implementation of the SubjectManager to did:nuts and did:web. They are not yet called by anyone.

### CODE REVIEW

File: vdr/didnuts/manager.go [hunk 1]
```diff
@@ -67,3 +95,193 @@ func (m Manager) UpdateService(_ context.Context, _ did.DID, _ ssi.URI, _ did.Se
...
+			},
+		},
+	}
+
+	return &sqlDoc, nil
+}
+
+func (m Manager) NewVerificationMethod(ctx context.Context, controller did.DID, _ didsubject.DIDKeyFlags) (*did.VerificationMethod, error) {
+	// did:nuts uses EC keys for everything so it doesn't use the DIDKeyFlags
+	return CreateNewVerificationMethodForDID(ctx, controller, m.KeyStore)
```
Comment (gerardsn) [inline vdr/didnuts/manager.go:143]: all dids are now self-controlled, so the term `controller` will become weirder over time. It's used several times in this file. Not sure if its worth changing though

File: vdr/didnuts/manager.go [hunk 2]
```diff
@@ -67,3 +95,193 @@ func (m Manager) UpdateService(_ context.Context, _ did.DID, _ ssi.URI, _ did.Se
...
+		for _, cik := range c.CapabilityInvocation {
+			key, err = m.KeyStore.Resolve(ctx, cik.ID.String())
+			if err == nil {
+				return c, key, nil
+			}
+		}
+	}
+
+	if errors.Is(err, nutsCrypto.ErrPrivateKeyNotFound) {
+		// log
```
Comment (gerardsn) [inline vdr/didnuts/manager.go:283]: todo

File: vdr/didnuts/manager.go [hunk 3]
```diff
@@ -67,3 +95,193 @@ func (m Manager) UpdateService(_ context.Context, _ did.DID, _ ssi.URI, _ did.Se
...
+type cryptoKey struct {
+	vm did.VerificationMethod
+}
+
+func (c cryptoKey) KID() string {
+	return c.vm.ID.String()
+}
+
+func (c cryptoKey) Public() crypto.PublicKey {
+	pk, _ := c.vm.PublicKey() // todo
```
Comment (gerardsn) [inline vdr/didnuts/manager.go:259]: todo

File: vdr/didweb/manager.go [hunk 4]
```diff
@@ -333,3 +338,74 @@ func buildDocument(newDID did.DID, doc sql.DIDDocument) (did.Document, error) {
...
+	verificationMethodID := did.DIDURL{
+		DID:      controller,
+		Fragment: uuid.New().String(),
+	}
+	if keyUsage.Is(didsubject.KeyAgreementUsage) {
+		return nil, errors.New("key agreement not supported for did:web")
+		// todo requires update to nutsCrypto module
+		//verificationMethodKey, err = m.keyStore.NewRSA(ctx, func(key crypt.PublicKey) (string, error) {
+		//	return verificationMethodID.String(), nil
+		//})
```
Comment (gerardsn) [inline vdr/didweb/manager.go:389]: is this part of the larger refactor or is there an issue for this?

**Review by gerardsn — CHANGES_REQUESTED** [review]

File: vdr/didweb/manager.go (hunk 4 above)
Comment (woutslakhorst) [reply vdr/didweb/manager.go:389]: #1948

**Review by gerardsn — APPROVED** [review]

**Pull request merged**